[23.2] Planemo serve does not show history panel

**Describe the bug**

`planemo serve` currently does not show me a history. Terminal shows: 
```
uvicorn.access INFO 2024-03-23 10:57:37,954 [pN:main.1,p:548391,tN:MainThread] 127.0.0.1:42354 - "GET /history/current_history_json HTTP/1.1" 500
[2024-03-23 10:57:37 +0100] [548391] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 428, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/fastapi/applications.py", line 284, in __call__
    await super().__call__(scope, receive, send)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette_context/middleware/raw_middleware.py", line 93, in __call__
    await self.app(scope, receive, send_wrapper)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/middleware/base.py", line 109, in __call__
    await response(scope, receive, send)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/responses.py", line 270, in __call__
    async with anyio.create_task_group() as task_group:
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 597, in __aexit__
    raise exceptions[0]
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/responses.py", line 273, in wrap
    await func()
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/middleware/base.py", line 134, in stream_response
    return await super().stream_response(send)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/responses.py", line 262, in stream_response
    async for chunk in self.body_iterator:
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/middleware/base.py", line 98, in body_stream
    raise app_exc
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/middleware/base.py", line 70, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/starlette/routing.py", line 443, in handle
    await self.app(scope, receive, send)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/a2wsgi/wsgi.py", line 157, in __call__
    return await responder(scope, receive, send)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/a2wsgi/wsgi.py", line 198, in __call__
    raise self.exc_info[0].with_traceback(
  File "/tmp/tmppxn4itnm/galaxy-dev/lib/galaxy/web/framework/middleware/error.py", line 165, in __call__
    app_iter = self.application(environ, sr_checker)
  File "/tmp/tmppxn4itnm/galaxy-dev/lib/galaxy/web/framework/middleware/remoteuser.py", line 201, in __call__
    return self.app(environ, start_response)
  File "/home/berntm/.planemo/gx_venv_3/lib/python3.10/site-packages/paste/httpexceptions.py", line 640, in __call__
    return self.application(environ, start_response)
  File "/tmp/tmppxn4itnm/galaxy-dev/lib/galaxy/web/framework/base.py", line 173, in __call__
    return self.handle_request(request_id, path_info, environ, start_response)
  File "/tmp/tmppxn4itnm/galaxy-dev/lib/galaxy/web/framework/base.py", line 262, in handle_request
    body = method(trans, **kwargs)
  File "/tmp/tmppxn4itnm/galaxy-dev/lib/galaxy/web/framework/decorators.py", line 64, in call_and_format
    rval = func(self, trans, *args, **kwargs)
  File "/tmp/tmppxn4itnm/galaxy-dev/lib/galaxy/web/framework/decorators.py", line 138, in set_nocache_headers
    return func(self, trans, *args, **kwargs)
  File "/tmp/tmppxn4itnm/galaxy-dev/lib/galaxy/webapps/galaxy/controllers/history.py", line 777, in current_history_json
    return self.history_data(trans, history)
  File "/tmp/tmppxn4itnm/galaxy-dev/lib/galaxy/webapps/galaxy/controllers/history.py", line 752, in history_data
    return self.history_serializer.serialize_to_view(history, view="dev-detailed", user=trans.user, trans=trans)
  File "/tmp/tmppxn4itnm/galaxy-dev/lib/galaxy/managers/base.py", line 783, in serialize_to_view
    return self.serialize(item, all_keys, **context)
  File "/tmp/tmppxn4itnm/galaxy-dev/lib/galaxy/managers/base.py", line 696, in serialize
    returned[key] = self.serializers[key](item, key, **context)
  File "/tmp/tmppxn4itnm/galaxy-dev/lib/galaxy/managers/histories.py", line 832, in serialize_contents_active
    return self.manager.contents_manager.active_counts(history)
  File "/tmp/tmppxn4itnm/galaxy-dev/lib/galaxy/managers/history_contents.py", line 180, in active_counts
    hda_select = self._active_counts_statement(model.HistoryDatasetAssociation, history.id)
AttributeError: 'NoneType' object has no attribute 'id'
```

I manually tried one commit before https://github.com/galaxyproject/galaxy/pull/17657 and the problem goes away.

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 23.2
Commit: https://github.com/galaxyproject/galaxy/pull/17657

**Browser and Operating System**
Operating System: Linux
Browser: Firefox

**To Reproduce**

`planemo serve`

--- a/lib/galaxy/webapps/base/webapp.py
+++ b/lib/galaxy/webapps/base/webapp.py
@@ -912,36 +912,28 @@ class GalaxyWebTransaction(base.DefaultWebTransaction, context.ProvidesHistoryCo
         Gets or creates a default history and associates it with the current
         session.
         """
+
+        # Just return the current history if one exists and is not deleted.
         history = self.galaxy_session.current_history
         if history and not history.deleted:
             return history
 
+        # Look for an existing history that has the default name, is not
+        # deleted, and is empty. If this exists, we associate it with the
+        # current session and return it.
         user = self.galaxy_session.user
         if user:
-            # Look for default history that (a) has default name + is not deleted and
-            # (b) has no datasets. If suitable history found, use it; otherwise, create
-            # new history.
             stmt = select(self.app.model.History).filter_by(
                 user=user, name=self.app.model.History.default_name, deleted=False
             )
             unnamed_histories = self.sa_session.scalars(stmt)
-            default_history = None
             for history in unnamed_histories:
                 if history.empty:
-                    # Found suitable default history.
-                    default_history = history
-                    break
+                    self.set_history(history)
+                    return history
 
-            # Set or create history.
-            if default_history:
-                history = default_history
-                self.set_history(history)
-            else:
-                history = self.new_history()
-        else:
-            history = self.new_history()
-
-        return history
+        # No suitable history found, create a new one.
+        return self.new_history()
 
     def get_most_recent_history(self):
         """

--- a/lib/galaxy/webapps/base/webapp.py
+++ b/lib/galaxy/webapps/base/webapp.py
@@ -938,6 +938,8 @@ class GalaxyWebTransaction(base.DefaultWebTransaction, context.ProvidesHistoryCo
                 self.set_history(history)
             else:
                 history = self.new_history()
+        else:
+            history = self.new_history()
 
         return history
 

--- a/lib/galaxy/webapps/base/webapp.py
+++ b/lib/galaxy/webapps/base/webapp.py
@@ -936,8 +936,8 @@ class GalaxyWebTransaction(base.DefaultWebTransaction, context.ProvidesHistoryCo
             if default_history:
                 history = default_history
                 self.set_history(history)
-        else:
-            history = self.new_history()
+            else:
+                history = self.new_history()
 
         return history
 

